resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,417 @@
++          }
++        }
+      }
+    }
+  }
+}
+```
+
+Step 5: Developer references the `Applications.Core/secretStores` resource in their `Applications.Datastores/*` or `Applications.Messaging/*` resource definition to securely manage secrets for use in their application.
+
+```diff
+resource db 'Applications.Datastores/mongoDatabases@2023-10-01-preview' = {
+  name: 'db'
+  properties: {
+    environment: environment
+    application: app.id
+    resourceProvisioning: 'manual'
+    host: substring(cosmosAccount.properties.documentEndpoint, 0, lastIndexOf(cosmosAccount.properties.documentEndpoint, ':'))
+    port: int(split(substring(cosmosAccount.properties.documentEndpoint,lastIndexOf(cosmosAccount.properties.documentEndpoint, ':') + 1), '/')[0])
+    database: cosmosAccount::database.name
+    username: ''
+    resources: [
+      { id: cosmosAccount.id }
+    ]
+    secrets: {
++        connectionString: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'connectionString'
++            }
++          }
++        }
++        password: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'password'
++            }
++          }
++        }
+    }
+  }
+}
+```
+
+```diff
+resource rabbitmq 'Applications.Messaging/rabbitmqQueues@2023-10-01-preview' = {
+  name: 'rabbitmq'
+  properties: {
+    environment: environment
+    application: app.id
+    resourceProvisioning: 'manual'
+    queue: 'radius-queue'
+    host: rmqHost
+    port: rmqPort
+    vHost: vHost
+    username: rmqUsername
+    secrets: {
++      password: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'password'
++            }
++          }
++      }
+    }
+  }
+}
+```
+
+Step 6: The developer specifies the `Applications.Core/secretStores` resource as a parameter to the Bicep resource that requires the secret.
+
+```bicep
+param secretStore object = authcreds
+
+resource demo 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:latest'
+      env:{
+        DB_CONNECTION: {
+          valueFrom: {
+            secretRef: {
+              source: secretStore.id
+              key: 'username'
+            }
+          }
+        }
+      }
+      ports: {
+        web: {
+          containerPort: 3000
+        }
+      }
+    }
+  }
+}
+```
+
+Step 7: Developer deploys the resources to Radius and the secrets required are either injected into the container as environment variables or used as credentials for authentication into the extender, volume, datastore, or messaging resource at deploy time. If the resource is deployed using a Recipe, the secrets are securely passed as outputs to Radius by the Recipe for use in provisioning and deploying the resource.
```

Comment:
I'm really confused about this walkthrough because the steps are numbered 1-7, but they appear unrelated, or to address different usecases.